src/MainWindow.vala

```diff
@@ -291,25 +241,16 @@ public class Installer.MainWindow : Hdy.Window {
     }
 
     private void load_success_view () {
-        if (success_view != null) {
-            success_view.destroy ();
-        }
-
-        success_view = new SuccessView ();
-        stack.add (success_view);
-        stack.visible_child = success_view;
+        var success_view = new SuccessView ();
+        deck.add (success_view);
+        deck.visible_child = success_view;
     }
 
     private void load_error_view (string log) {
-        if (error_view != null) {
-            error_view.destroy ();
-        }
-
-        error_view = new ErrorView (log);
-        stack.add (error_view);
-        stack.visible_child = error_view;
-
-        error_view.previous_view = disk_view;
+        var error_view = new ErrorView (log);
+        deck.add (error_view);
+        deck.visible_child = error_view;
+        deck.can_swipe_back = true;
```

Comment:
This should be false? Otherwise it is possible to swipe back to progress view and further back.